code-experiments/build/java/ExampleExperimentNonAnytime.java

```diff
@@ -0,0 +1,233 @@
+     *
+     * @param suiteName Name of the suite (e.g. "bbob", "bbob-biobj", or "bbob-constrained").
+     * @param observerName Name of the observer matching with the chosen suite (e.g. "bbob-biobj"
+     * when using the "bbob-biobj-ext" suite).
+     * @param randomGenerator The random number generator.
+     */
+    public static void exampleExperimentNonAnytime(String suiteName, String observerName, Random randomGenerator) {
+        try {
+            final String algorithm_name = "RS";  // no spaces allowed
+            final String output_folder = algorithm_name;  // no spaces allowed
+
+            /* Initialize the suite.
+             * For more details on how to change the default options, see
+             * http://numbbo.github.io/coco-doc/C/#suite-parameters. */
+            Suite suite = new Suite(suiteName, "", "");
+
+            // Set some options for the observer. See documentation for other options.
+            final String observerOptions = "result_folder: " + output_folder + " algorithm_name: " + algorithm_name;
+            /* Initialize the observer.
+             * For more details on how to change the default options, see
+             * http://numbbo.github.io/coco-doc/C/#observer-parameters. */
+            Observer observer = new Observer(observerName, observerOptions);
+
+            // A list of increasing budgets to be multiplied by dimension
+            // gradually increase `max_budget` to 10, 100, ...
+            // or replace with a user-defined list
+            int[] budgetMultiplierList = defaultBudgetList(5, 50);
+            System.out.println("Benchmarking with budgets: "+Arrays.toString(budgetMultiplierList)+" (* dimension)");
+
+            Benchmark benchmark = new Benchmark(suite, observer);
+
+            // Iterate over all problems in the suite
+            while ((PROBLEM = benchmark.getNextProblem()) != null) {
+
+                int dimension = PROBLEM.getDimension();
+
+                for(int b : budgetMultiplierList){
+
+                    long evaluationsDone = PROBLEM.getEvaluations() + PROBLEM.getEvaluationsConstraints();
+                    long evaluationsRemaining = (long) dimension * b - evaluationsDone;
+
+                    /* Break the loop if the target was hit or there are no more remaining evaluations */
+                    if (PROBLEM.isFinalTargetHit() || (evaluationsRemaining <= 0))
+                        break;
+
+                    /* Call the optimization algorithm for the remaining number of evaluations */
+                    MyRandomSearch(evaluateFunction,
+                            dimension,
+                            PROBLEM.getNumberOfObjectives(),
+                            PROBLEM.getNumberOfConstraints(),
+                            PROBLEM.getSmallestValuesOfInterest(),
+                            PROBLEM.getLargestValuesOfInterest(),
+                            evaluationsRemaining,
+                            randomGenerator);
+
```

Comment:
What is missing here is the reset of the problem in order to prepare logging for the next budget like in the non-anytime [python code](https://github.com/numbbo/coco/blob/master/code-experiments/build/python/example_experiment_non_anytime.py#L110), or am I wrong?